server/cli/src/load_test.rs

```diff
@@ -218,93 +371,226 @@ impl LoadTest {
+                        println!("Failed to spawn process: {}", e);
+                        return;
+                    }
+                };
+                sleep(Duration::from_secs(10)).await; // Let db get created, migrated and initialisation started
+                let client = Client::new();
+
+                if test_site.wait_for_sync().await.is_err() {
+                    kill(&mut child).await;
+                    return;
+                }
+
+                let start = std::time::Instant::now();
+                loop {
+                    let requisition_id = uuid();
+                    let requisition_gql = json!({
+                        "operationName": "InsertRequestRequisition",
+                        "query": INSERT_REQUISITION_MUTATION,
+                        "variables": {
+                            "storeId": test_site.store.id,
+                            "input": {
+                                "id": requisition_id,
+                                "otherPartyId": test_site.next_store.name_id,
+                                "maxMonthsOfStock": 3,
+                                "minMonthsOfStock": 1
+                            }
+                        }
+                    });
+                    match test_site.do_post(&requisition_gql).await {
+                        Ok(response) => response,
+                        Err(e) => {
+                            println!("insertRequestRequisition request failed: {}", e);
+                            kill(&mut child).await;
+                            return;
+                        }
+                    };
+
+                    for i in 0..num_lines {
+                        let line_id = uuid();
+                        let line_gql = json!({
+                            "operationName": "InsertRequestRequisitionLine",
+                            "query": INSERT_REQUISITION_LINE_MUTATION,
+                            "variables": {
+                                "storeId": test_site.store.id,
+                                "input": {
+                                    "id": line_id,
+                                    "itemId": item_ids_copy[i%num_lines],
+                                    "requisitionId": requisition_id
+                                }
+                            }
+                        });
+
+                        match test_site.do_post(&line_gql).await {
+                            Ok(response) => response,
+                            Err(e) => {
+                                println!("insertRequestRequisitionLine request failed: {}", e);
+                                kill(&mut child).await;
+                                return;
+                            }
+                        };
+
+                        let line_gql = json!({
+                            "operationName": "UpdateRequestRequisitionLine",
+                            "query": UPDATE_REQUISITION_LINE_MUTATION,
```

Comment:
curious once we start adding measurements, but adding each line as insert+update rather than adding all lines via one batch mutation... _could_ make the OMS site the bottleneck rather than sync? I mean, if it is, that's good to know - but if we're going for load test against OG, maybe the fewer set up requests we have to do to the remote site the better?

Reply:
Honestly I just found the individual update graphql first and went with it- figured surely it's not significantly slower when the requests are originating from the same machine (i.e. the CLI and spawned processes, not the same as central servers ideally).

I would rather do the batch insert and update but was rolling with it for now :S. If nothing else it'll make hyper's debug logging quieter (25x less in fact) which would be pleasant lol.